pkg/skaffold/jib/jib.go

```diff
@@ -78,3 +100,33 @@ func getDependencies(cmd *exec.Cmd) ([]string, error) {
 	sort.Strings(deps)
 	return deps, nil
 }
+
+// calculateRoots returns a list of possible symlink-expanded paths
+func calculateRoots(path string) ([]string, error) {
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to resolve %s", path)
+	}
+	canonical, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to canonicalize workspace %s", path)
+	}
+	if path == canonical {
+		return []string{path}, nil
+	}
+	return []string{canonical, path}, nil
+}
+
+// relativize tries to make path relative to one of the given roots
+func relativize(path string, roots ...string) (string, error) {
+	if !filepath.IsAbs(path) {
```

Comment:
Why are we assuming here that if the path is relative, it is relative to the right root? Can we assume that?